### PR TITLE
fix: report merge issues

### PR DESCRIPTION
**Pull request by nickofthyme**
I made a few more changes in addition to #2 related to errors merging multiple report outputs.

It was over a month ago so I don't recall specifics but now that this is package is showing new signs of life I figured I would add the changes I made if you care for them.

To the best of memory this fixes:
- Issue with merging the base64 encoded `data/` files across many reports including those within the same test suite. This included merging the running stats.
- Issue with merging reports with no failures, hence no `data/` files.
- Issue with finding the `index.html` file from `playwright-core`. This appears to have been fixed #3 but I think a better approach is just to use the `index.html` file that we already have while merging the reports.

Happy to accept changes and answer questions.

I committed the `dist/` under [`nickofthyme/playwright-merge-html-reports#latest-temp`](https://github.com/nickofthyme/playwright-merge-html-reports/tree/latest-temp), so you can test out these changes using...

```jsonc
// package.json
{
  "dependencies": {
    "@playwright/test": "^1.21.1",
    "playwright-merge-html-reports": "nickofthyme/playwright-merge-html-reports#latest-temp"
  } 
}
```

### CODE REVIEW

File: src/merge-reports.ts [hunk 1]
```diff
@@ -37,28 +39,23 @@ async function mergeHTMLReports(inputReportPaths: string[], givenConfig: Config
 
   // Merge config with default values
   const { outputFolderName, outputBasePath } = {...defaultConfig, ...givenConfig};
-
-  const b64StartSearchStr = `window.playwrightReportBase64 = "data:application/zip;base64,`;
-  const b64EndSearchStr = `";</script>`;
```
Comment (nickofthyme) [inline src/merge-reports.ts:42]: Replaced with `RegExp` here 👇🏼 

https://github.com/anooprav7/playwright-merge-html-reports/blob/1b7e94cac450d086e16f9c1d32c62db45f7275f8/src/merge-reports.ts#L32

**Comment by mstepin** [comment]
This appears to have fixed most issues for us. Looking forward to it being part of the official release version. Thanks @nickofthyme

**Comment by nickofthyme** [comment]
@mstepin thanks for confirming, in the meantime I will keep [`nickofthyme/playwright-merge-html-reports#latest-temp`](https://github.com/nickofthyme/playwright-merge-html-reports/tree/latest-temp) as is if you want to use it. I will only change/delete it once this has been merged/released into the official package for a few weeks.

**Comment by nickofthyme** [comment]
@anooprav7 Please have a look at this at your earliest convenience.

**Comment by mrmckeb** [comment]
Thanks for the work on this @nickofthyme!

**Comment by anooprav7** [comment]
Merged and published - [playwright-merge-html-reports@0.2.3](https://www.npmjs.com/package/playwright-merge-html-reports/v/0.2.3). Thank you @nickofthyme

**Comment by mrmckeb** [comment]
After this PR, this no longer works for us. We're seeing:
```
Error: Unhandled error: Error: ENOENT: no such file or directory, open '/home/runner/work/[path]/playwright-report/index.html'
```

For now we've pinned to `0.2.2`.

**Comment by mrmckeb** [comment]
I'm trying to debug this now @nickofthyme and have a question:

> Issue with finding the index.html file from playwright-core. This appears to have been fixed https://github.com/anooprav7/playwright-merge-html-reports/pull/3 but I think a better approach is just to use the index.html file that we already have while merging the reports.

What is the `index.html` file you're referring to? Is this supposed to be the first of the reports to merge?

**Comment by nickofthyme** [comment]
Hey @mrmckeb thanks for bringing this up. The `index.html` I am referring to is the one generated by the html reporter that contains the base64 encoded and zipped data. Let me take a look at it today.

Would you be able to share the reports that you are attempting to merge?

**Comment by nickofthyme** [comment]
First off I could not reproduce the error you posted above. Could you triple check that you are running the `v0.2.3`, show output from `npm ls playwright-merge-html-reports`. Likely you are using it but just want to be sure before digging deeper.

I put up a branch to test any issues [`#fix-index-issue`](https://github.com/nickofthyme/playwright-merge-html-reports/tree/fix-index-issue). If you could add an example set of reports to this branch I can fix any errors.

---

To elaborate on your questions above, the changes I made consider two cases of report output, the successful case and the failure case.

The successful report, see [`html_report-4/`](https://github.com/nickofthyme/playwright-merge-html-reports/tree/fix-index-issue/html_report-4), has a folder structure in this case is just the `index.html` file that.

```
html_report-4
└── index.html

0 directories, 1 file
```

The failure report, see [`html_report-5/`](https://github.com/nickofthyme/playwright-merge-html-reports/tree/fix-index-issue/html_report-5), has a folder structure shown below.

```
html_report-5
├── data
│   ├── 28020256a2448811325e73ba1f4e43d860a9f078.png
│   ├── 3894dbf4890cb9d13bb4432c357166317f8fdcab.zip
│   ├── 44ae4426e4ed12f2946cce749264aafa893d3f06.png
│   └── a60fa73165275bc5ecc091b38c78c060ae9ebfa9.webm
├── index.html
└── trace
    ├── codicon.dcd00fb4.ttf
    ├── icon-16x16.png
    ├── icon-192x192.png
    ├── icon-256x256.png
    ├── icon-32x32.png
    ├── icon-384x384.png
    ├── icon-512x512.png
    ├── index.html
    ├── main.a5d7e729.css
    ├── main.ce7ea3d1.js
    └── sw.bundle.js

2 directories, 16 files
```

So we start at the first report with the `index.html` file, which should always exist AFAIK, and extract the data from `window.playwrightReportBase64`. The content of which contains a list of files corresponding to each test grouping along with the `report.json` which contains the stats. See the test group `nick.test.ts` in the html report below is linked to the `fileId` of `78b2aedcc60c5dc5a20bed4bd29ccfdec2697364`.

Playwright Test Report 2022-06-16 at 9 03 02 AM


```json5
// html_report-5
{
  files: [Object: null prototype] {
    '78b2aedcc60c5dc5a20bed4bd29ccfdec2697364.json': {
      name: '78b2aedcc60c5dc5a20bed4bd29ccfdec2697364.json',
      dir: false,
      date: 2022-06-16T15:08:40.000Z,
      comment: null,
      unixPermissions: 33204,
      dosPermissions: null,
      _data: [Object],
      _dataBinary: true,
      options: [Object]
    },
    'report.json': {
      name: 'report.json',
      dir: false,
      date: 2022-06-16T15:08:40.000Z,
      comment: null,
      unixPermissions: 33204,
      dosPermissions: null,
      _data: [Object],
      _dataBinary: true,
      options: [Object]
    }
  },
  comment: null,
  root: '',
  clone: [Function (anonymous)]
}
```

The contents of the `<fileID>.json` files contain the test info which is merged across other reports referring to the same test group/`fileId`. So you can see all the test info below, this info may point to assets inside of the `data/` directory that is copied over to the merged report directory.

```json5
{
  fileId: '78b2aedcc60c5dc5a20bed4bd29ccfdec2697364',
  fileName: 'nick.test.ts',
  tests: [
    {
      testId: '78b2aedcc60c5dc5a20bed4bd29ccfdec2697364-5de94c3ac94f0ab6c0a8eb221a2bfa9d6d884ccd',
      title: 'Testing hover with body',
      projectName: 'chrome',
      location: [Object],
      duration: 3456,
      annotations: [],
      outcome: 'unexpected',
      path: [Array],
      results: [Array],
      ok: false
    },
    {
      testId: '78b2aedcc60c5dc5a20bed4bd29ccfdec2697364-019368faaefea77fed6d2a88a877da6754bb61af',
      title: 'Testing hover with root',
      projectName: 'chrome',
      location: [Object],
      duration: 3473,
      annotations: [],
      outcome: 'expected',
      path: [Array],
      results: [Array],
      ok: true
    },
    {
      testId: '78b2aedcc60c5dc5a20bed4bd29ccfdec2697364-3d902908fbc1b8dff027c301765e2e759b2ca1ef',
      title: 'Testing hover with chart',
      projectName: 'chrome',
      location: [Object],
      duration: 3343,
      annotations: [],
      outcome: 'expected',
      path: [Array],
      results: [Array],
      ok: true
    }
  ]
}
```

---

With all that said, I don't see how the `index.html` would throw an error that you showed. So example reports from you would be ideal.

I realize now that I missed copying over the `trace/` directory with includes having to destruct a `.zip` file in the `data/` which I can also look into.

**Comment by mrmckeb** [comment]
Hey @nickofthyme, sorry for the slow reply here - we had a long weekend and I took a little time away from the keyboard.

When testing it seems that the new code expects a root HTML file. The reports are pulled in to dirs like `/report-1` and `/report-2`, but there isn't a root `index.html` file for the module to merge into.

Maybe it wasn't being copied over correctly as you identified.

**Comment by nickofthyme** [comment]
Hey @mrmckeb, no worries!

Ok good to know, I found an issue in the source copy logic that I fixed in #6. Have you had a chance to test that version?

**Comment by mrmckeb** [comment]
Too many threads, sorry!

**Pull request merged**